### PR TITLE
Issue 1370 b

### PR DESCRIPTION
**Pull request by terfloth**

### CODE REVIEW

File: test-plugins/org.yakindu.sct.generator.java.test/src-gen/org/yakindu/scr/choice/ChoiceStatemachine.java [hunk 1]
```diff
@@ -349,6 +349,7 @@ private void react_main_region_A() {
 				} else {
 					if (check_main_region_A_tr3_tr3()) {
 						effect_main_region_A_tr3();
+					} else {
```
Comment (andreasmuelder) [inline test-plugins/org.yakindu.sct.generator.java.test/src-gen/org/yakindu/scr/choice/ChoiceStatemachine.java]: Seems that empty elses are generated now (at least in Java)

File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/TimedTransitions.java [hunk 2]
```diff
@@ -26,24 +26,32 @@
...
 	public void timer01() {
 		statemachine.enter();
 		assertTrue(statemachine.isStateActive(State.main_region_Start));
-		timer.timeLeap(2030);
+		for(int i = 1; i <= 10; i++) {
+			timer.timeLeap(200);
+			statemachine.runCycle();
+		}
+		timer.timeLeap(30);
+		statemachine.runCycle();
```
Comment (andreasmuelder) [inline test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/TimedTransitions.java]: This is not right to trigger a runcycle after timeleap that is not a multiple of the cycle period

Comment (terfloth) [inline test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/TimedTransitions.java]: wow -  who put that piece of smart code generator code into place?!?

File: test-plugins/org.yakindu.sct.generator.java.test/src-gen/org/yakindu/scr/choice/ChoiceStatemachine.java (hunk 1 above)
Comment (terfloth) [inline test-plugins/org.yakindu.sct.generator.java.test/src-gen/org/yakindu/scr/choice/ChoiceStatemachine.java]: is  fixed with the new commit

File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/TimedTransitions.java (hunk 2 above)
Comment (terfloth) [inline test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/TimedTransitions.java]: out of scope for this pull request ...

**Pull request merged**